### PR TITLE
origin modal filter items

### PR DESCRIPTION
**Pull request by ryan-mahoney**
#### Summary of changes
**Asana Ticket:** [⏱Schedules | Input box - Stops filter textbox](https://app.asana.com/0/555089885850811/1122431599573150)

Screen Shot 2019-06-20 at 11 47 00 AM

<br>
Assigned to: @meagonqz

### CODE REVIEW

File: apps/site/assets/ts/components/Modal.tsx [hunk 1]
```diff
@@ -48,44 +61,21 @@ const ModalContent = ({
...
-    // Deactivate
-    return () => {
-      if (bodyWrapper) {
-        bodyWrapper.setAttribute("aria-hidden", "false");
-        htmlElement.classList.remove("modal-open");
-        bodyWrapper.style.paddingRight = bodyPadding;
-      }
-      trap.deactivate();
-    };
-  });
```
Comment (meagonqz) [inline apps/site/assets/ts/components/Modal.tsx]: If you add a dependency array of [] here (as the last arg to the hook) then I think it solves your problem without getting rid of the focus trap and adding extra logic.

You will want to do something like ` focusElementId={state.modalId === "origin" ? "origin-search" : ...}`

Comment (ryan-mahoney) [reply apps/site/assets/ts/components/Modal.tsx]: incredible. Thank you so much for pointing this out!

**Comment by ingridpierre** [comment]
image

Changes requested:
- Bold for title
- Since the search isn't a search it's a filter it would be nice to change the placeholder to "Filter stops and stations"
- Since selecting required, "Select from the list below" is more appropriate than "Or select.."
- Can we make just the div with the stop list scroll rather than the entire modal? I want to draw more attention to the filter since people tended to miss it before.

**Comment by ingridpierre** [comment]
image
- Intermittently seeing the modal ignore my direction selection. In other words, the stop list sometimes shows the stops for the wrong direction.

**Comment by ryan-mahoney** [comment]
I figured out the sorting issue: the component needs to take into consideration the default direction sorting that is set on the controller level -- fixed.

**Pull request merged**